govcd/vapp.go

```diff
@@ -190,18 +197,27 @@ func (vapp *VApp) AddNewVM(name string, vappTemplate VAppTemplate, network *type
 		AllEULAsAccepted: acceptAllEulas,
 	}
 
+	// Add storage profile
+	if storageProfileRef != nil && storageProfileRef.HREF != "" {
+		vAppComposition.SourcedItem.StorageProfile = storageProfileRef
+	}
+
 	// Inject network config
-	vcomp.SourcedItem.InstantiationParams.NetworkConnectionSection = network
+	vAppComposition.SourcedItem.InstantiationParams.NetworkConnectionSection = network
 
 	apiEndpoint, _ := url.ParseRequestURI(vapp.VApp.HREF)
 	apiEndpoint.Path += "/action/recomposeVApp"
 
 	// Return the task
 	return vapp.client.ExecuteTaskRequest(apiEndpoint.String(), http.MethodPost,
-		types.MimeRecomposeVappParams, "error instantiating a new VM: %s", vcomp)
+		types.MimeRecomposeVappParams, "error instantiating a new VM: %s", vAppComposition)
 
 }
 
+// ========================= issue#252 ==================================
```

Comment:
Again maybe a TODO would work here or a note in the issue.

Reply:
Done